replica/database.hh

```diff
@@ -1347,6 +1347,7 @@ private:
 
     flat_hash_map<sstring, keyspace> _keyspaces;
     std::unordered_map<table_id, lw_shared_ptr<column_family>> _column_families;
+    std::unordered_map<sstring, lw_shared_ptr<column_family>> _disposed_column_families;
```

Comment:
Storing a shared pointer here means the table object is never released, until someone triggers a `clean_disposed_column_families()`. AFAIK currently only attempting to create/remove a table can trigger this.

A better solution is to store a `weak_ptr<column_family>` here. This would not prevent the table from being destroyed when all its users are gone, while still providing a way to check if the table is alive or not.

Reply:
Alternative proposal: add removed tables to an auto-unlink intrusive map. Destoyed tables would automatically remove themselves from this, so we wouldn't even need `clean_disposed_column_families()`.

Reply:
Right, if tables removed themselves from this list (don't even need weird intrusive maps), you won't need clean_displosed_column_families. Also, you wouldn't need to keep the actual table object in this map - if you want to prevent creation of this table *name*, you can just keep these names, not the full table objects.